pipeline_tools/shared/metadata_utils.py

```diff
@@ -0,0 +1,141 @@
+from pipeline_tools.shared import dcp_utils
+from humancellatlas.data.metadata.api import Bundle, CellSuspension
+from pipeline_tools.shared.http_requests import HttpRequests
+import functools
+from concurrent.futures import ThreadPoolExecutor
+
+
+def get_bundle_metadata(uuid, version, dss_url, http_requests):
+    """Factory function to create a `humancellatlas.data.metadata.Bundle` object from bundle information and manifest.
+
+    Args:
+        bundle_uuid (str): The bundle uuid.
+        bundle_version (str): The bundle version.
+        dss_url (str): Url of Data Storage System to query
+        http_requests (HttpRequests): An HttpRequests object.
+
+    Returns:
+        humancellatlas.data.metadata.Bundle: A bundle metadata object.
+    """
+    manifest = dcp_utils.get_manifest(
+        bundle_uuid=uuid,
+        bundle_version=version,
+        dss_url=dss_url,
+        http_requests=http_requests,
+    )['bundle']['files']
+
+    metadata_files_dict = {f['name']: f for f in manifest if f['indexed']}
+    metadata_files = get_metadata_files(
+        metadata_files_dict=metadata_files_dict, dss_url=dss_url
+    )
+
+    return Bundle(
+        uuid=uuid, version=version, manifest=manifest, metadata_files=metadata_files
+    )
+
+
+def get_sample_id(bundle):
+    """Return the sample id from the given bundle.
+
+    Args:
+        bundle (humancellatlas.data.metadata.Bundle): A Bundle object contains all of the necessary information.
+
+    Returns:
+        sample_id (str): String giving the sample id
+    """
+    sample_id = str(bundle.sequencing_input[0].document_id)
+    return sample_id
+
+
+def get_expected_cell_count(bundle):
```

Comment:
This is only used by cellranger, so I'd say this can go in `cellranger.py`

Reply:
That's a good point, yeah we should move anything that's used by only one pipeline closer to that pipeline lib to prevent getting this file too large! Will do.